app/components/ITunesCard/index.js

```diff
@@ -0,0 +1,157 @@
+/**
+ *
+ * ITunesCard
+ *
+ */
+
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import truncate from 'lodash/truncate';
+import { isEmpty } from 'lodash';
+import { Card, Popover, Button, Progress } from 'antd';
+import { PlayCircleOutlined, PauseCircleOutlined } from '@ant-design/icons';
+import T from '@components/T';
+import If from '@components/If';
+import { colors } from '@app/themes';
+import { ACTIONS } from '@utils/constants';
+import logo from '@images/icon-512x512.png';
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 1.25em 0;
+    box-shadow: ${colors.boxShadow};
+  }
+`;
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin-bottom: 0.75rem;
+  gap: 1em;
+`;
+const CustomImg = styled.img`
+  display: block;
+  margin: 0.75rem auto;
+  height: 10rem;
+`;
+
+export function ITunesCard({ track, onActionClick }) {
+  const audioRef = useRef(null);
+  const [progress, setProgress] = useState(0);
+  const [intervalStore, setIntervalStore] = useState(null);
```

Comment:
Why should this be a state variable ?

Reply:
I'm using a setInterval to calculate progress of song for `<Progress/>` component.
I believe we should clear such timers to reduce unnecessary memory consumption. So storing it in a state so that i can clear it using `clearInterval(intervalStore)`

Reply:
won't variable scoped above the component work as well ?

Reply:
Yes, that should work as well